File: LogisimPlus/LogisimPlus/logisim/gui/prefs/OptionsPanel.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.prefs
{

	internal abstract class OptionsPanel : JPanel
	{
		private PreferencesFrame optionsFrame;

		public OptionsPanel(PreferencesFrame frame) : base()
		{
			this.optionsFrame = frame;
		}

		public OptionsPanel(PreferencesFrame frame, LayoutManager manager) : base(manager)
		{
			this.optionsFrame = frame;
		}

		public abstract string Title {get;}

		public abstract string HelpText {get;}

		public abstract void localeChanged();

		internal virtual PreferencesFrame PreferencesFrame
		{
			get
			{
				return optionsFrame;
			}
		}
	}

}

File: logisim/app/src/main/java/logisim/gui/prefs/OptionsPanel.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.prefs;

import javax.swing.JPanel;

abstract class OptionsPanel extends JPanel {
	private PreferencesFrame optionsFrame;

	public OptionsPanel(PreferencesFrame frame) {
		super();
		optionsFrame = frame;
	}

	public abstract String getTitle();

	public abstract String getHelpText();

	public abstract void localeChanged();

	PreferencesFrame getPreferencesFrame() {
		return optionsFrame;
	}
}
